src/item_factory.cpp

```diff
@@ -2608,31 +2608,97 @@ void Item_factory::add_special_pockets( itype &def )
     }
 }
 
-static int parse_acc_string( const std::string &to_hit )
+namespace acc_parse
+{
+// *INDENT-OFF*
+enum class grip_val : int {
+    BAD = -1,
+    NONE,
+    SOLID,
+    WEAPON
+};
+enum class length_val : int {
+    HAND = 0,
+    SHORT,
+    LONG
+};
+enum class surface_val : int {
+    POINT = -2,
+    LINE,
+    ANY,
+    EVERY
+};
+enum class balance_val : int {
+    CLUMSY = -2,
+    UNEVEN,
+    NEUTRAL,
+    GOOD
+};
```

Comment:
so normally we don't explicitly define these (all of the enums), i would in this instance just to make sure they're exactly the value you intend them to be. In addition, each of these needs a LAST member, and then enum_traits defined as well, like so:
```
template<>
struct enum_traits<acc_parse::grip_val> {
    static constexpr auto last = acc_parse::grip_val::LAST;
};
```

Reply:
`enum_to_string` and `string_to_enum` won't work for enums that don't start at 0.

Reply:
Just needed to reset the enum base to 0.

Fixed.